Replace $container `has` check with `hasParameter`

Buratinas opened the pull request:
When extensions are being loaded one by one into tmpContainer inside
`vendor/symfony/dependency-injection/Compiler/MergeExtensionConfigurationPass.php`
`sonata_admin` configuration is loaded alone, and other extension services are not present.
By replacing `has` check with `hasParameter` check we can make sure parameter exists instead of service.
Since we can easily configure extra parameter,
but to get service we would need to load most of the acl bundle extension services.

<!-- THE PR TEMPLATE IS NOT AN OPTION. DO NOT DELETE IT, MAKE SURE YOU READ AND EDIT IT! -->
## Subject

<!-- Describe your Pull Request content here -->

<!--
    Show us you choose the right branch.
    Different branches are used for different things :
    - 4.x is for everything backwards compatible, like patches, features and deprecation notices
    - 5.x is for deprecation removals and other changes that cannot be done without a BC-break
    More details here: https://github.com/sonata-project/SonataAdminBundle/blob/4.x/CONTRIBUTING.md#base-branch
-->
I am targeting this branch, because I'm working with this and it's the latest branch.

<!--
    Specify which issues will be fixed/closed.
    Remove it if this is not related.
-->

Closes #7768.

## Changelog

```markdown
### Fixed
- Improve detection of Symfony ACL bundle installation
```

Buratinas committed:
Replace $container `has` check with `hasParameter`

When extensions are being loaded one by one into tmpContainer inside
`vendor/symfony/dependency-injection/Compiler/MergeExtensionConfigurationPass.php`
`sonata_admin` configuration is loaded alone, and other extension services are not present.
By replacing `has` check with `hasParameter` check we can make sure parameter exists instead of service.
Since we can easily configure extra parameter,
but to get service we would need to load most of the acl bundle extension services.

jordisala1991 commented on src/DependencyInjection/SonataAdminExtension.php:
```diff
@@ -144,7 +144,7 @@ public function load(array $configs, ContainerBuilder $container): void
                 break;
             case 'sonata.admin.security.handler.acl':
-                if (!$container->has('security.acl.provider')) {
+                if (!$container->hasParameter('security.acl.provider')) {
```
why don't we instead check for the presence of the bundle enabled?, something like:

```suggestion
                if (!isset($bundles['AclBundle'])) {
```

Buratinas replied:
that actually might work better. I was checking bundles in compiler passes and extensions and it never came to my mind to simply replace this check with bundle check, although I wonder if that's the only bundle which can do ACL. I'll add it in front and I guess I can leave this as fallback since it should work with simple or statement.

jordisala1991 replied:
Keep in mind that the only integration we support is that bundle, We use some interfaces from there and dependencies also declared there. I wouldn't compplicate it any further, to me we should remove other checks and leave only the bundle one, wdyt @VincentLanglet ?

Buratinas replied:
I thought this might be the case. If support is only with `symfony/acl-bundle`, then bundle check and error message change should be enough - if it only checks if bundle exists, then the message should probably not mention `security.acl.provider` at all, just the part with required bundle.

Buratinas committed:
Only check for bundle for fail fast

We only need to fail fast if:
- bundle does not exist

This means if we use AclBundle we will succeed this check.

jordisala1991 commented:
Thank you @Buratinas
